results/cmd/api/main_test.go

```diff
@@ -63,3 +63,27 @@ func setupTestDB(dbName string, t *testing.T) (*sql.DB, error) {
 	}
 	return db, nil
 }
+
+func TestDeleteTaskRun(t *testing.T) {
+	// Create a temporay database
+	testDB, err := setupTestDB("testdb", t)
+	if err != nil {
+		t.Fatalf("failed to create temp file for db: %v", err)
+	}
+
+	// connect to fake server and do testing
+	srv := &server{db: testDB}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	objectmeta := &pb.ObjectMeta{Uid: "123459", Name: "mytaskrun", Namespace: "default"}
+	r, err := srv.CreateTaskRun(ctx, &pb.CreateTaskRunRequest{TaskRun: &pb.TaskRun{ApiVersion: "1", Metadata: objectmeta}})
+	if err != nil {
+		t.Fatalf("could not create taskrun: %v", err)
+	}
+	res, err := srv.DeleteTaskRun(ctx, &pb.DeleteTaskRunRequest{Uid: "123459"})
```

Comment:
Is this UID generated by the server? If so we should derive the parameter values from the returned task above so that the test will be less brittle to changes to the UID generation.

Reply:
done, I used r.GetMetadata().GetUid() instead